dm/worker/server.go

```diff
@@ -254,8 +254,10 @@ func (s *Server) doStartKeepAlive() {
 }
 
 func (s *Server) stopKeepAlive() {
-	s.kaCancel()
-	s.kaWg.Wait()
+	if s.kaCancel != nil {
```

Comment:
in which case `stopKeepAlive` is called before or concurrent with `startKeepAlive`? we may need a lock if so 😂

Reply:
1. dm-worker/main.go `s.Start()`
2. Start() return an error before `s.startKeepAlive()`. e.g.: worker port already in use
3. dm-worker/main.go `s.Close()`, report a nil pointer info in stdout